### PR TITLE
feat: Add horizontal and vertical alignment options to Column

### PR DESCRIPTION
**Pull request by VincentSmedinga**
- Adds the same alignment properties to Column as #1330 did for Row.
- Improves the examples in both Row and Column 
  - Visualizes the flex container through a striped background
  - Sets its width and height for better impression of alignment use cases
  - Simplifies the class names used
  - Adds JSDoc and a separate type to optimize Storybook Controls
  - ~~Hides other props from specific examples – the Default example will always have all props~~

### CODE REVIEW

File: packages/react/src/Column/Column.tsx [hunk 1]
```diff
@@ -6,22 +6,49 @@
...
 type ColumnGap = (typeof columnGapSizes)[number]
+type ColumnTag = 'article' | 'div' | 'section'
 
 export type ColumnProps = {
-  /** The HTML element to use. */
+  /**
+   * The vertical alignment of the items in the column.
+   * @default start
+   */
+  align?: MainAlign
```
Comment (alimpens) [inline packages/react/src/Column/Column.tsx:21]: Can you add tests for these new properties as well?

File: storybook/src/components/Column/Column.docs.mdx [hunk 2]
```diff
@@ -26,9 +23,40 @@ To add white space below a single element, you can alternatively use a utility c
...
+<Canvas of={ColumnStories.Alignment} />
+
+### Horizontal alignment
+
+Items in the column can be aligned horizontally in several ways:
+
+- **Start**: align items to the start of the column. This is the left side in left-to-right languages.
+- **Center**: center items within the column.
+- **End**: align items to the end of the column. This is the right side in left-to-right languages.
+- **Start**: align items to the left of the column.
```
Comment (alimpens) [inline storybook/src/components/Column/Column.docs.mdx:50]:
```suggestion
```

File: storybook/src/components/Column/Column.stories.tsx [hunk 3]
```diff
@@ -32,34 +54,57 @@ type Story = StoryObj<typeof meta>
...
 
+export const Alignment: Story = {
+  args: {
+    align: 'center',
+  },
+  argTypes: {
+    alignHorizontal: { table: { disable: true } },
+    as: { table: { disable: true } },
+    gap: { table: { disable: true } },
+  },
```
Comment (alimpens) [inline storybook/src/components/Column/Column.stories.tsx:65]: Why did you disable these controls here? We generally allow all controls for all stories.

File: storybook/src/components/Row/Row.stories.tsx [hunk 4]
```diff
@@ -3,43 +3,47 @@
...
+      options: [undefined, ...mainAlignOptions],
     },
     alignVertical: {
-      control: 'radio',
-      options: crossAlignOptions,
+      control: {
+        type: 'radio',
+        labels: { undefined: 'stretch' },
+      },
+      options: [undefined, ...crossAlignOptions.filter((option) => option !== 'baseline')],
```
Comment (alimpens) [inline storybook/src/components/Row/Row.stories.tsx:36]: Why is `baseline` filtered from the options here? You can't recreate the 'Align opposing texts'-story this way.

File: storybook/src/components/Row/Row.stories.tsx [hunk 5]
```diff
@@ -50,41 +54,85 @@ type Story = StoryObj<typeof meta>
...
 export const AlignOpposingTexts: Story = {
   args: {
     align: 'between',
     alignVertical: 'baseline',
     children: [<Heading level={3}>An example heading</Heading>, <Link href="#">An example link</Link>],
+    className: undefined,
     wrap: true,
   },
+  argTypes: {
+    alignVertical: { table: { disable: true } },
```
Comment (alimpens) [inline storybook/src/components/Row/Row.stories.tsx:106]: This story is about vertical alignment, no?

File: storybook/src/components/Row/Row.stories.tsx (hunk 4 above)
Comment (VincentSmedinga) [reply storybook/src/components/Row/Row.stories.tsx:36]: Fixed – may have mixed up Row and Column halfway.

File: storybook/src/components/Row/Row.stories.tsx (hunk 5 above)
Comment (VincentSmedinga) [reply storybook/src/components/Row/Row.stories.tsx:106]: Hah :)

File: storybook/src/components/Column/Column.stories.tsx (hunk 3 above)
Comment (VincentSmedinga) [reply storybook/src/components/Column/Column.stories.tsx]: They’re all back.

**Review by alimpens — APPROVED** [review]

**Pull request merged**